core/tests/protractor_utils/ExplorationPlayerPage.js

```diff
@@ -0,0 +1,164 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Page object for the exploration player, for use in Protractor
+ * tests.
+ */
+
+var forms = require('./forms.js');
+var general = require('./general.js');
+var interactions = require('../../../extensions/interactions/protractor.js');
+
+var ExplorationPlayerPage = function() {
+  var conversationInputAll =
+    element.all(by.css('.protractor-test-conversation-input'));
+  var conversationInput =
+    element(by.css('.protractor-test-conversation-input'));
+  var conversationContent =
+    element.all(by.css('.protractor-test-conversation-content'));
+  var conversationFeedback =
+    element.all(by.css('.protractor-test-conversation-feedback'));
+  var toNextCard =
+    element(by.css('.protractor-test-continue-to-next-card-button'));
+  var feedbackPopupLink =
+    element(by.css('.protractor-test-exploration-feedback-popup-link'));
+  var suggestionPopupLink =
+    element(by.css('.protractor-test-exploration-suggestion-popup-link'));
+  var feedbackTextArea =
+    element(by.css('.protractor-test-exploration-feedback-textarea'));
+  var feedbackSubmitButton =
+    element(by.css('.protractor-test-exploration-feedback-submit-btn'));
+  var explorationHeader =
+    element(by.css('.protractor-test-exploration-header'));
+  var ratingStar = element.all(by.css('.protractor-test-rating-star'));
+  var explorationInfoIcon =
+    element(by.css('.protractor-test-exploration-info-icon'));
+  var infoCardRating = element(by.css('.protractor-test-info-card-rating'));
+  var hint = element(by.css('.protractor-test-view-hint'));
+  var solution = element(by.css('.protractor-test-view-solution'));
+  var restart =
```

Comment:
restartExplorationButton

That said, I didn't know we had a restart exploration button. Is this actually used? If not, please remove.

Reply:
Ya, its not being used, will remove that.

Reply:
Done